noir-projects/aztec-nr/aztec/src/macros/functions/mod.nr

```diff
@@ -35,10 +44,13 @@ use auth_registry::AUTHORIZE_ONCE_REGISTRY;
 pub comptime fn initializer(f: FunctionDefinition) {
     // Marker attribute - see the comment above
 
-    if !is_fn_private(f) & !is_fn_public(f) {
+    if !is_fn_external(f) {
```

Comment:
This is the main sad thing about this PR. Now I am unable to evaluate here if the function is not utility because I don't have the guarantee if the external macro function has already been run. So I only perform this check here and then in utility macro I check if these attributes are not applied.

Reply:
I would go with the second approach as it will make the macros much clearer.

The only bad thing about it is that invalid attribute args currently don't play well with LSP:

<img width="2040" height="1712" alt="image" src="https://github.com/user-attachments/assets/4ba4fee3-7ab2-4f8b-8390-3f7ff543811c" />

But I assume that Noir team could improve that and give us a nice error message at which point all seems fine.